src/containers/location/LocationsSagas.js

```diff
@@ -514,18 +522,129 @@ function* searchLocationsWorker(action :SequenceAction) :Generator<any, any, any
     LOG.error(action.type, error);
     yield put(searchLocations.failure(action.id));
   }
-
-  return response;
+  finally {
+    yield put(searchLocations.finally(action.id));
+  }
 }
 
 function* searchLocationsWatcher() :Generator<any, any, any> {
   yield takeEvery(SEARCH_LOCATIONS, searchLocationsWorker);
 }
 
+function* searchReferralAgenciesWorker(action :SequenceAction) :Generator<any, any, any> {
+  const response = {
+    data: {}
+  };
+
+  try {
+    yield call(refreshAuthTokenIfNecessary);
+
+    const { value } = action;
+    const { selectedOption } = value;
+    if (!isPlainObject(value)) throw ERR_ACTION_VALUE_TYPE;
+
+    let latLonObj = selectedOption;
+    if (!isImmutable(latLonObj)) latLonObj = fromJS(latLonObj);
+
+    yield put(searchReferralAgencies.request(action.id, {
+      searchInputs: latLonObj
+    }));
+
+    const latitude :string = get(latLonObj, 'lat');
+    const longitude :string = get(latLonObj, 'lon');
+
+    if (isFunction(gtag)) {
+      gtag('event', 'Execute Search', {
+        event_category: 'Search',
+      });
+    }
+
+    const entitySetId = RR_ENTITY_SET_ID;
+    let app :Map = yield select((state) => state.get('app', Map()));
+    let providerEntitySetId = getProvidersESID(app);
+
+    if (!providerEntitySetId) {
+      const loadAppRequest = loadApp();
+      yield put(loadAppRequest);
+      yield takeReqSeqSuccessFailure(loadApp, loadAppRequest);
+      app = yield select((state) => state.get('app', Map()));
+      providerEntitySetId = getProvidersESID(app);
+    }
+
+    const locationPropertyTypeId = getPropertyTypeId(app, PROPERTY_TYPES.LOCATION);
+
+    const sort = {
+      type: 'geoDistance',
+      descending: false,
+      propertyTypeId: locationPropertyTypeId,
+      latitude,
+      longitude
+    };
+
+    const locationConstraint = {
+      constraints: [{
+        type: 'geoDistance',
+        latitude,
+        longitude,
+        propertyTypeId: locationPropertyTypeId,
+        radius: 100,
+        unit: 'miles'
+      }, {
+        type: 'simple',
+        fuzzy: false,
+        searchTerm: `_exists_:entity.${locationPropertyTypeId}`
+      }],
+      min: 2
+    };
+
+    const constraints = [locationConstraint];
+
+    const searchOptions = {
+      start: 0,
+      entitySetIds: [entitySetId],
+      maxHits: 5,
+      constraints,
+      sort
+    };
+
+    const { hits, numHits, error } = yield call(SearchApi.searchEntitySetData, searchOptions);
+    if (error) throw error;
+
+    const filteredHits = fromJS(hits).filter((e) => e.get(PROPERTY_TYPES.LOCATION, List()).size);
```

Comment:
No change required, just wondering why you didn't use `has` instead?

```
import { has } from 'immutable';
const filteredHits = fromJS(hits).filter((e) => has(e, PROPERTY_TYPES.LOCATION));
```

Alternatively, I think you could also use the safer `getPropertyValue`:

```
const filteredHits = fromJS(hits).filter((e) => getPropertyValue(e, PROPERTY_TYPES.LOCATION, List()).size);
```

Reply:
using `has` is the better option

See above discussion on `getPropertyValue`
https://github.com/openlattice/child-care/pull/147#discussion_r527275168

Reply:
This whole thing came from the searchLocationsWorker which I didn't write. It's my bad for no refactoring after I got it up in running. Will change.